src/domain/chains/chains.repository.spec.ts

```diff
@@ -0,0 +1,302 @@
+import type { Page } from '@/domain/entities/page.entity';
+import type { ILoggingService } from '@/logging/logging.interface';
+
+const mockLoggingService = {
+  error: jest.fn(),
+} as jest.MockedObjectDeep<ILoggingService>;
+const mockConfigApi = {
+  getChains: jest.fn(),
+} as jest.MockedObjectDeep<IConfigApi>;
+const mockTransactionApiManager =
+  {} as jest.MockedObjectDeep<ITransactionApiManager>;
+const mockConfigurationService = jest.mocked({
+  getOrThrow: jest.fn(),
+} as jest.MockedObjectDeep<IConfigurationService>);
+
+/**
+ * Note: all other methods of the repository are tested in situ.
+ * Whilst `getAllChains` is to some extent as well, the following
+ * tests are required to cover all edge cases.
+ */
+describe('ChainsRepository', () => {
+  // According to the limits of the Config Service
+  // @see https://github.com/safe-global/safe-config-service/blob/main/src/chains/views.py#L14-L16
+  const OFFSET = 40;
+
+  let target: ChainsRepository;
+  const maxSequentialPages = 3;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+
+    mockConfigurationService.getOrThrow.mockImplementation((key) => {
+      if (key === 'safeConfig.chains.maxSequentialPages')
+        return maxSequentialPages;
+    });
+
+    target = new ChainsRepository(
+      mockLoggingService,
+      mockConfigApi,
+      mockTransactionApiManager,
+      mockConfigurationService,
+    );
+  });
+
+  it('should return all chains across pages below request limit', async () => {
+    const url = faker.internet.url({ appendSlash: true });
+    const chains = Array.from(
+      {
+        length: ChainsRepository.MAX_LIMIT * (maxSequentialPages - 1), // One page less than request limit
+      },
+      (_, i) => chainBuilder().with('chainId', i.toString()).build(),
+    );
+    const pages = chunk(chains, ChainsRepository.MAX_LIMIT).map(
+      (results, i, arr) => {
+        const pageOffset = (i + 1) * OFFSET;
+
+        const previous = ((): string | null => {
+          if (i === 0) {
+            return null;
+          }
+          return limitAndOffsetUrlFactory(
+            ChainsRepository.MAX_LIMIT,
+            pageOffset,
+            url,
+          );
+        })();
+        const next = ((): string | null => {
+          if (i === arr.length - 1) {
+            return null;
+          }
+          return limitAndOffsetUrlFactory(
+            ChainsRepository.MAX_LIMIT,
+            pageOffset,
+            url,
+          );
+        })();
+
+        return pageBuilder<Chain>()
+          .with('results', results)
+          .with('count', chains.length)
+          .with('previous', previous)
+          .with('next', next)
+          .build();
+      },
+    );
+    mockConfigApi.getChains.mockImplementation(
+      ({ offset }): Promise<Page<Chain>> => {
+        if (offset === 0) {
+          return Promise.resolve(pages[0]);
+        }
+        if (offset === 40) {
```

Comment:
What about using:
`offset === offset + ChainsRepository.MAX_LIMIT`
`offset === offset + ChainsRepository.MAX_LIMIT * 2`
This way, if the max limit is changed in the future, the tests won't fail.

However, it might also be intentional for the tests to break if the limit changes, as it could signal that the change needs further attention.

Reply:
Changed to use constant in 3b4b46c25b4d282d29daa09f485730ab65e2dc65.